packages/nodes-base/nodes/GraphQL/GraphQL.node.ts

```diff
@@ -327,239 +598,573 @@ export class GraphQL implements INodeType {
 	};
 
 	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
-		const items = this.getInputData();
-		let httpBasicAuth;
-		let httpDigestAuth;
-		let httpCustomAuth;
-		let httpHeaderAuth;
-		let httpQueryAuth;
-		let oAuth1Api;
-		let oAuth2Api;
-
-		try {
-			httpBasicAuth = await this.getCredentials('httpBasicAuth');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			httpCustomAuth = await this.getCredentials('httpCustomAuth');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			httpDigestAuth = await this.getCredentials('httpDigestAuth');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			httpHeaderAuth = await this.getCredentials('httpHeaderAuth');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			httpQueryAuth = await this.getCredentials('httpQueryAuth');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			oAuth1Api = await this.getCredentials('oAuth1Api');
-		} catch (error) {
-			// Do nothing
-		}
-		try {
-			oAuth2Api = await this.getCredentials('oAuth2Api');
-		} catch (error) {
-			// Do nothing
-		}
+		const connectionMode = this.getNodeParameter('connectionMode', 0, 'http') as string;
 
-		let requestOptions: IRequestOptions;
+		if (connectionMode === 'websocket') {
```

Comment:
Is this really needed here or can there be checks in the existing loop to keep the code a bit clearer?